### PR TITLE
Show the username as a fallback on feeds if full name is blank

### PR DESCRIPTION
**Pull request by jamesorlakin**
While the `DEFAULT_SHOW_FULL_NAME` works well, there's a bug on the feed/dashboard page that causes the username not be used as a fallback and nothing shown instead:
image

This PR copies the logic from `DisplayName` in `models/user.go:741` to return the username as a fallback for the template:
image
_(where `review` is the username, and `Jimbo` as a user with the full name set)_

I'm happy to make a backport too. 😃

### CODE REVIEW

**Review by zeripath — COMMENTED** [review]
Not that we're the most efficient system but perhaps it would be better to restructure this so we only do the trimspace if the setting is set.

**Comment by jamesorlakin** [comment]
To be honest I'm not really sure what the trim is there for. I guess if the user puts in a full name of just spaces we should ignore it, but surely that validation should be done at setting time?

**Comment by jamesorlakin** [comment]
@zeripath String trimming in both the user and action models are now done only if `DEFAULT_SHOW_FULL_NAME` is configured.

**Review by zeripath — APPROVED** [review]

File: models/action.go [hunk 1]
```diff
@@ -120,10 +120,13 @@ func (a *Action) ShortActUserName() string {
...
 }
 
-// GetDisplayName gets the action's display name based on DEFAULT_SHOW_FULL_NAME
+// GetDisplayName gets the action's display name based on DEFAULT_SHOW_FULL_NAME, or falls back to the username if it is blank.
 func (a *Action) GetDisplayName() string {
 	if setting.UI.DefaultShowFullName {
-		return a.GetActFullName()
+		trimmedFullName := strings.TrimSpace(a.GetActFullName())
+		if len(trimmedFullName) > 0 {
+			return a.GetActFullName()
```
Comment (lafriks) [inline models/action.go]:
```suggestion
			return trimmedFullName
```

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/action.go (hunk 1 above)
Comment (jamesorlakin) [reply models/action.go]: Doh! 🤦‍♂

**Review by 6543 — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**